components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt

```diff
@@ -29,9 +29,10 @@ data class DownloadState(
     val contentLength: Long? = null,
     val userAgent: String? = null,
     val destinationDirectory: String = Environment.DIRECTORY_DOWNLOADS,
-    val filePath: String =
-        Environment.getExternalStoragePublicDirectory(destinationDirectory).path + "/" + fileName,
     val referrerUrl: String? = null,
     val skipConfirmation: Boolean = false,
     val id: Long = Random.nextLong()
-)
+) {
+    val filePath: String get() =
```

Comment:
We should move the local filePath out of the state here, but let's do this as part of https://github.com/mozilla-mobile/android-components/pull/5047/files#r347619081.